sentinel-adapter/sentinel-apache-dubbo-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/dubbo/SentinelDubboConsumerFilter.java

```diff
@@ -98,24 +101,26 @@ private Result syncInvoke(Invoker<?> invoker, Invocation invocation) {
         }
     }
 
-
     private Result asyncInvoke(Invoker<?> invoker, Invocation invocation) {
         LinkedList<EntryHolder> queue = new LinkedList<>();
         String prefix = DubboAdapterGlobalConfig.getDubboConsumerResNamePrefixKey();
         String interfaceResourceName = getInterfaceName(invoker, prefix);
         String methodResourceName = getMethodName(invoker, invocation, prefix);
         try {
-            queue.push(new EntryHolder(SphU.asyncEntry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT), null));
-            queue.push(new EntryHolder(SphU.asyncEntry(methodResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT, 1, invocation.getArguments()), invocation.getArguments()));
+            queue.push(new EntryHolder(
+                SphU.asyncEntry(interfaceResourceName, ResourceTypeConstants.COMMON_RPC, EntryType.OUT), null));
+            queue.push(new EntryHolder(
+                SphU.asyncEntry(methodResourceName, ResourceTypeConstants.COMMON_RPC,
+                    EntryType.OUT, 1, invocation.getArguments()), invocation.getArguments()));
             Result result = invoker.invoke(invocation);
-            result.whenCompleteWithContext(new BiConsumer<Result, Throwable>() {
-                @Override
-                public void accept(Result result, Throwable throwable) {
-                    while (!queue.isEmpty()) {
-                        EntryHolder holder = queue.pop();
-                        Tracer.traceEntry(result.getException(), holder.entry);
-                        exitEntry(holder);
-                    }
+            result.whenCompleteWithContext((r, throwable) -> {
+                while (!queue.isEmpty()) {
+                    EntryHolder holder = queue.pop();
+                    Throwable error = Optional.ofNullable(r)
+                        .map(Result::getException)
+                        .orElse(throwable);
```

Comment:
Maybe the priority list is `throwable` -> `r.exception` indicating `throwable` is more fatal. How do you think about it?

Reply:
Updated, please check.